### ISSUE TITLE
compatibility for IR compiler

### ISSUE THREAD

**Issue by CreatsheR** [gciatto/kt-math#5]
As the title says. I've seen there is a branch for it, but there wasn't any activity since one month.
Is there a reason, why it isn't merged into master yet? We would like to use the library, but without IR compatibility we can't.

**Comment by gciatto** [comment]
I was experimenting with the new IR on that branch, but I was considering it a low-priority task.
There is no blocking issue TBH. I'm simply quite busy at work, currently, and I'm therefore very slow with low-priority tasks.

However, once Kt `1.5.30` is out, the new IR will be considered in the beta stage, and this will increase the priority of the porting.
I hope I'll be able to complete the transition in the next weeks or months.
Apologies for slowing down your workflow.

**Comment by danwallach** [comment]
Plus-one for this. I'm currently working on a project that needs to use a number of libraries. It seems that it's something of a mess if you want to use "legacy" libraries and "IR" libraries together. If you could publish "both" to Maven, then I think this headache would go away for me.

**Comment by gciatto-unibo** [comment]
Yeah combining IR- and legacy-compiled libraries is not supported.

Long story short: I'm waiting for Kt `1.6.20` to be released as [this issue](https://youtrack.jetbrains.com/issue/KT-45434) (which is currently preventing me from switching to the new IR) should be fixed by that version
